android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -260,6 +260,15 @@ private WebResourceResponse handleCapacitorHttpRequest(WebResourceRequest reques
             headers.put(header.getKey(), header.getValue());
         }
 
+        // a workaround for the following android web view issue:
+        // https://issues.chromium.org/issues/40450316
+        // x-cap-user-agent contains the user agent set in JavaScript
+        String userAgentValue = headers.getString("x-cap-user-agent");
+        if (userAgentValue != null && !userAgentValue.isEmpty()) {
```

Comment:
```suggestion
        if (userAgentValue != null) {
```
I noticed iOS allows empty strings, so we should also allow them for Android